hedera-node/src/main/java/com/hedera/services/store/contracts/HederaWorldState.java

```diff
@@ -69,37 +73,35 @@
 	private final EntityIdSource ids;
 	private final HederaLedger ledger;
 	private final ServicesRepositoryRoot repositoryRoot;
-
 	private final List<Address> provisionalAccountDeletes = new ArrayList<>();
 	private final Map<Address, Bytes> provisionalCodeUpdates = new HashMap<>();
 	private final Map<Address, WorldStateAccount> provisionalAccountUpdates = new HashMap<>();
 	private final Map<Address, Address> sponsorMap = new LinkedHashMap<>();
+	private final LedgerAccountsSource ledgerAccountsSource;
 
 	@Inject
 	public HederaWorldState(
 			final EntityIdSource ids,
 			final HederaLedger ledger,
-			final ServicesRepositoryRoot repositoryRoot
-	) {
+			final ServicesRepositoryRoot repositoryRoot,
+			final LedgerAccountsSource ledgerAccountsSource
+			) {
 		this.ids = ids;
-		this.ledger = ledger;
 		this.repositoryRoot = repositoryRoot;
+		this.ledger = ledger;
+		this.ledgerAccountsSource = ledgerAccountsSource;
 	}
 
-
 	@Override
-	public List<ContractID> persist() {
+	public List<ContractID> persist(Optional<Address> creator) {
 		provisionalAccountDeletes.forEach(address -> {
-			final var accountId = accountParsedFromSolidityAddress(address.toArray());
 			// account may have been created and deleted within the same transaction.
-			if (ledger.exists(accountId)) {
-				var merkleAccount = ledger.get(accountId);
+			if (repositoryRoot.isExist(address.toArray())) {
+				var account = repositoryRoot.getAccount(address.toArray());
```

Comment:
`address.toArray()` should be stored in a variable as it may re-create a byte array each call and we use it multiple tiems.